src/client/search/searchActions.js

```diff
@@ -98,6 +103,59 @@ const receiveFeedResults = (data, requestId, recievedFromTwitter) => ({
   fetchedRequestFromTwitter: recievedFromTwitter,
 });
 
+export const INVALIDATE_JOURNALISM_INFORMATION = 'INVALIDATE_JOURNALISM_INFORMATION';
+export const REQUEST_ENTITY = 'REQUEST_ENTITY';
+export const RECEIVE_ENTITY = 'RECEIVE_ENTITY';
+export const invalidateJournalismInfo = () =>
+  (dispatch, getState) => {
+    let invalidated = false;
+    getState().searchTerms.forEach(searchTerm => {
+      if (searchTerm.entity) {
+        // only requests the entity from dbpedia if it hasn't already been fetched.
+        if (!getState().journalismInfo.entities[searchTerm.id]) {
+          // only invalidates journalismInfo if there is an entity in the search terms list
+          // and it has not already been invalidated.
+          if (!invalidated) {
+            dispatch({ type: INVALIDATE_JOURNALISM_INFORMATION });
+            invalidated = true;
+          }
+
+          dispatch({
+            type: REQUEST_ENTITY,
+            id: searchTerm.id,
+            query: searchTerm.query,
+            entityType: searchTerm.entity,
+          });
+
+          // hits different end points depending on whether it's a player or team being queried.
+          switch (searchTerm.entity) {
+            case PLAYER_ENTITY:
+              newPromiseChain()
+                .then(() => fetch(`/journalism/players/${searchTerm.query}`, makeGetHeader()))
+                .then(response => response.json())
+                .then()
```

Comment:
but also fixed in my current version :S can we pull then I'm opening up a merge straight away without it anyway